dashboard/test/helpers/delete_accounts_helper_test.rb

```diff
@@ -1584,6 +1584,20 @@ def get_record_with_sql(select_attribute, table_name, selector)
     assert_removes_field_from_form_geos :longitude
   end
 
+  #
+  # Table: dashboard.ai_tutor_interactions
+  #
+
+  test "deletes all of a soft-deleted user's ai tutor interactions (chat messages)" do
+    student = create :student_with_ai_tutor_access
+    num_ai_tutor_interactions = 3
+    create_list :ai_tutor_interaction, num_ai_tutor_interactions, user: student
+
+    assert_changes -> {AiTutorInteraction.where(user: student).count}, from: num_ai_tutor_interactions, to: 0 do
+      student.destroy
```

Comment:
I believe that `User#destroy` doesn't trigger the `DeleteAccountsHelper#purge_user` which calls `DeleteAccountsHelper#delete_ai_tutor_interactions`, so it won't work
```suggestion
      purge_user(student)
```
https://github.com/code-dot-org/code-dot-org/blob/5e6b5c62d0975bcc54b4573091c1634c2148b441/dashboard/test/helpers/delete_accounts_helper_test.rb#L2185-L2202

Reply:
@artem-vavilov is right that `purge_user` is only called when the user is hard-deleted after 30 days of being soft-deleted. @Erin007 I think there is a disconnect between the PR description and the tests written. Do you want the AI messages to be deleted when the user is soft-deleted or when the user is hard-deleted?

If you want the messages to be deleted when the user is hard-deleted, then the code in this PR is good, but the PR description should be updated to reflect that.

Reply:
Thanks for the clarification @daynew! I was missing that nuance. I updated the PR title and description as well as test name to reflect that we are indeed deleting the chat message records when the user is purged.